portal/api-authentication-grpc.go

```diff
@@ -19,12 +19,94 @@ package portal
 import (
 	"context"
 
+	"database/sql"
+	"fmt"
+
 	"golang.org/x/crypto/bcrypt"
 
 	cluster "github.com/minio/m3/cluster"
 	pb "github.com/minio/m3/portal/stubs"
+	"github.com/minio/minio-go/v6"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
 )
 
+type User struct {
+	Tenant   string
+	Email    string
+	IsAdmin  bool
+	Password string
+	UUID     string
+}
+
+func (s *server) MakeBucket(ctx context.Context, in *pb.MakeBucketRequest) (res *pb.Bucket, err error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, status.New(codes.InvalidArgument, "Metadata not found").Err()
+	}
+
+	var sessionId string
+	switch sIds := md.Get("sessionId"); len(sIds) {
+	case 0:
+		return nil, status.New(codes.Unauthenticated, "SessionId not found").Err()
+	default:
+		sessionId = sIds[0]
+	}
+
+	// With validating sessionId behind us, we query the tenant MinIO
+	// service corresponding to the logged-in user to make the bucket
+
+	// Prepare DB instance
+	db := cluster.GetInstance().Db
```

Comment:
Can you check `cluster.MakeBucket()` I think that function already does most of what is implemented here

Reply:
Yes, let me check that. On a related note, minio-go make bucket errors out because it is not able to resolve tenant address on the local development setup.

Reply:
(put the other comment on the wrong thread) that is because you need to port-forward the MinIO service to your local and add it to your `/etc/hosts`, alternatively you can use `kubefwd`, once M3 is running inside k8s it will have direct access to the services

Reply:
added a pr #75 on how to use kubefwd while developing

Reply:
> Can you check `cluster.MakeBucket()` I think that function already does most of what is implemented here

done